src/main/java/com/groupon/jenkins/DockerComposeDotCi/DockerPublishPlugin.java

```diff
@@ -76,48 +76,87 @@ public boolean perform(DynamicBuild build, Launcher launcher, BuildListener list
         return true;
     }
 
+    private ShellCommands PublishImages(ShellCommands cmds, String composeBuild, String composeImage, String ciTag,
+                                        String sha, String orgNrepo, boolean forcePushLatest) {
+
```

Comment:
What happens if forcePushLastest is true, and one of the two logic blocks (83-86 or 88-91) have already run?